Update default payment request button size to 48px

james-allan opened the pull request:
Fixes #2781

## Changes proposed in this Pull Request:

This PR updates the default Express payment button height from 40px to 48px. Stores will have their pre-existing settings maintained. For instance, if the store had the default (40px) set, it will be changed to `small`. If they they had `medium` (48px) it will be changed to default. 

| Before | After |
|--------|--------|
| <img width="450" alt="Screenshot 2023-11-20 at 2 45 33 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/33af2bd5-a33f-46e3-8cec-edd515f29c7a"> | <img width="450" alt="Screenshot 2023-11-20 at 2 41 49 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/fdabcdeb-b7e8-44eb-b50b-4a3a97b4f16c"> |

## Testing instructions

Zip of this branch for testing: [woocommerce-gateway-stripe.zip](https://github.com/woocommerce/woocommerce-gateway-stripe/files/13406647/woocommerce-gateway-stripe.zip)

> [!NOTE]
> I've bumped the **`WC_STRIPE_VERSION`** in that zip to make sure the migrate flow is triggered when you upload it. 

### Fresh site

- Create a new Jurassic Ninja site with WooCommerce installed.
- Upload the zip of the Stripe plugin (provided above).
- Go to the Stripe settings and connect or enter API keys. 
- Go to the _Payment Methods_ Stripe tab. 
- **Customize** the Apple Pay / Google Pay settings. 
- Note the button size is set to the middle option - **Default (48px)**. 

</p>
<img width="250" alt="Screenshot 2023-11-20 at 3 09 21 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/07db4edb-f1a8-4d8b-a71e-936b104488be">
</p>


### Existing site

- Create a new Jurassic Ninja site with WooCommerce.
- Install the live version of Stripe from the WP plugin WordPress.org installer.
- Go to the Stripe settings and connect or enter API keys.
- Go to the _Payment Methods_ Stripe tab. 
- **Customize** the Apple Pay / Google Pay settings. 
- Note the button size is set to 40px (previous default). 
- Upload the zip of the Stripe plugin (provided above).
- Go back to the Payment button settings page. 
- Note that the button size has remained 40px but it is labeled **"Small"**.
- Repeat for **"Medium"** or **"Large"**.
    - Medium (48px) will be changed to Default (48px).
    - Large will be unchanged. 

### General tests

- Changing the button size and saving, saves the button size.
- Create a simple product.
- Confirm the selected button size is reflected when viewing the product, cart or checkout page. 

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

james-allan committed:
Update default payment button size to 48px

james-allan committed:
Add changelog entry

james-allan committed:
Add migration flow of stores updating to 7.8

james-allan committed:
Update js test to expect new default

james-allan committed:
Add unit test for migration

james-allan committed:
Add unit tests for getting button height

james-allan committed:
Fix error in unit tests

diegocurbelo reviewed (approved):
Code looks good, and tests well, both the attached zip in JN and the dev brach tested locally. 🚢

james-allan commented:
@elizaan36 are you happy with this one too? I've summarised the changes I made below:

1. Changed **~~Default (40px)~~** → **Small (40px)**
2. Changed **~~Medium (48px)~~** → **Default (48px)**

| Before | After |
|--------|--------|
| <img width="450" alt="Screenshot 2023-11-20 at 2 45 33 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/33af2bd5-a33f-46e3-8cec-edd515f29c7a"> | <img width="450" alt="Screenshot 2023-11-20 at 2 41 49 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/fdabcdeb-b7e8-44eb-b50b-4a3a97b4f16c"> |

This PR also includes an upgrade routine to make sure existing sites keep their previously selected button height. ie if they had the previous default (40px), they will keep 40px (small).

elizaan36 commented:
Thanks @james-allan. LGTM!

james-allan committed:
Merge branch 'develop' into fix/2781-default-express-button-size
